lib/matplotlib/backend_bases.py

```diff
@@ -2012,6 +2029,13 @@ def print_figure(self, filename, dpi=None, facecolor=None, edgecolor=None,
             A list of extra artists that will be considered when the
             tight bbox is calculated.
 
+        backend : str, optional
+            When set, forcibly use this non-interactive backend for rendering.
+            This can be either a standard backend name ("agg", "pdf", "pgf",
+            "ps", "svg") or a custom backend name ("module://...").  The main
+            use case of this parameter is to render to a format using a backend
+            that is neither the current backend, nor the default backend for
+            this format, e.g. for rendering pdfs using the pgf backend.
```

Comment:
I'm still not clear.  Are there other examples, like cairo instead of agg etc?  If there is just one use case, I don't think this should get such a generic sounding kwarg name.  If there are other cases, I'm confused about which ones are allowed.  If I save to png, can I use pgf?  Can I use the pdf backend to save to svg?  Where are the compatibilities documented?

Reply:
Indeed, you can also use cairo (either the builtin, so "cairo", or mplcairo ("module://mplcairo.base")) to save to any format.
For the builtin backends the compat table is at https://matplotlib.org/devdocs/tutorials/introductory/usage.html#the-builtin-backends (except that pgf is missing -- will open a separate PR for that).

Reply:
OK, thats more clear.  I wonder if we change the kwarg to `renderer` and refer to this table somehow? 

```
renderer: str, optional
     Use a non-default backend to render the file, i.e. `renderer='cairo'` for a png rather than the default 'agg' backend, or 'pgf' for a PDF instead of the default 'pdf'.  See :doc:`/tutorials/introductory/usage` for a list of valid backends for each file format.  Note custom backends can be referenced as "module://...".
```

Reply:
BTW just to be clear, I'm balking at "backend" because the user will already have likely been subjected to the confusing idea that they have to select a backend, and then see this kwarg and assume the have to make it match up somehow.  I think we should try and be clear that 99.9% of the time the user should not need to use this kwarg.

Reply:
The doc sounds good, mostly used your suggestion with minor edits.
The argument matches the argument to `matplotlib.use()`, so I think it makes sense to keep the same argument name, so I'd still prefer `backend` here.

Reply:
and added "Note that the default backend is normally sufficient."

Reply:
Sounds good (though you didn't push anything yet, right?)

Reply:
oops, missed the push, done now